core/kazoo_speech/src/asr/kazoo_asr_google.erl

```diff
@@ -18,20 +18,42 @@
 -define(GOOGLE_ASR_KEY, kapps_config:get_binary(?GOOGLE_CONFIG_CAT, <<"asr_api_key">>, <<"">>)).
 -define(GOOGLE_ASR_PROFANITY_FILTER, kapps_config:get_is_true(?GOOGLE_CONFIG_CAT, <<"asr_profanity_filter">>)).
 -define(GOOGLE_ASR_ENABLE_WORD_TIME_OFFSETS, kapps_config:get_is_true(?GOOGLE_CONFIG_CAT, <<"asr_enable_word_time_offsets">>)).
+-define(GOOGLE_ASR_ENCODING, kapps_config:get_binary(?GOOGLE_CONFIG_CAT, <<"asr_encoding">>, <<"LINEAR16">>)).
+-define(GOOGLE_ASR_SAMPLE_RATE_HERTZ, kapps_config:get_binary(?GOOGLE_CONFIG_CAT, <<"asr_sample_rate_hertz">>, <<"16000">>)).
+-define(GOOGLE_ASR_MODEL, kapps_config:get_binary(?GOOGLE_CONFIG_CAT, <<"asr_model">>, <<"phone_call">>)).
+-define(GOOGLE_ASR_USE_ENHANCED, kapps_config:get_is_true(?GOOGLE_CONFIG_CAT, <<"asr_use_enhanced">>, 'true')).
+-define(GOOGLE_ASR_ENABLE_AUTOMATIC_PUNCTUATION, kapps_config:get_is_true(?GOOGLE_CONFIG_CAT, <<"asr_enable_automatic_punctuation">>, 'true')).
+
+-define(DEFAULT_ASR_CONTENT_TYPE, <<"application/wav">>).
+-define(SUPPORTED_CONTENT_TYPES, [<<"application/wav">>]).
+
 
 -spec commands(kz_term:ne_binary(), kz_term:ne_binaries(), kz_term:ne_binary(), kz_term:ne_binary(), kz_term:proplist()) -> provider_return().
 commands(_Bin, _Commands, _ContentType, _Locale, _Opts) ->
     {'error', 'asr_provider_failure', <<"Not implemented">>}.
 
+
 -spec freeform(binary(), kz_term:ne_binary(), kz_term:ne_binary(), kz_term:proplist()) -> asr_resp().
-freeform(Content, _ContentType, Locale, Options) ->
+freeform(Content, ContentType, Locale, Options) ->
+    case maybe_convert_content(Content, ContentType) of
+        {'error', _}=E -> E;
+        {Content1, ContentType1} -> exec_freeform(Content1, ContentType1, Locale, Options)
+    end.
+
+-spec exec_freeform(binary(), kz_term:ne_binary(), kz_term:ne_binary(), kz_term:proplist()) -> asr_resp().
+exec_freeform(Content, _ContentType, Locale, Options) ->
     BaseUrl = ?GOOGLE_ASR_URL,
     Headers = req_headers(),
     lager:debug("sending request to ~s", [BaseUrl]),
 
     AudioConfig = [{<<"languageCode">>, Locale}
                   ,{<<"profanityFilter">>, ?GOOGLE_ASR_PROFANITY_FILTER}
                   ,{<<"enableWordTimeOffsets">>, ?GOOGLE_ASR_ENABLE_WORD_TIME_OFFSETS}
+		,{<<"encoding">>, ?GOOGLE_ASR_ENCODING}
+		,{<<"sampleRateHertz">>, ?GOOGLE_ASR_SAMPLE_RATE_HERTZ}
```

Comment:
Think need use sample rate from audio file. Not static value.
I suggest to remove this, because Google engine will get sample rate from received wav file.